packages/client/screens/AuthenticationSignUpForm/index.tsx

```diff
@@ -0,0 +1,489 @@
+    createMachine<Record<string, never>, UpdateNicknameMachineEvent>({
+        id: 'Sign up',
+        initial: 'Idle',
+        states: {
+            Idle: {
+                on: {
+                    'Validated sign up form': {
+                        target: '#Sign up.Sending sign up to server',
+                    },
+                },
+            },
+
+            "Debouncing user's information fetching": {
+                after: {
+                    '300': {
+                        target: '#Sign up.Idle',
+                    },
+                },
+            },
+
+            'Sending sign up to server': {
+                invoke: {
+                    src: 'Send sign up to server',
+                },
+            },
+        },
+
+        on: {
+            'Signed up successfully': {
+                actions: ['Trigger sucess toast', 'Reset forms errors'],
+                target: '#Sign up.Idle',
+            },
+            'Nickname unavailable': {
+                actions: 'Set error state to nickname unavailable',
+                target: '#Sign up.Idle',
+            },
+            'Email unavailable': {
+                actions: 'Set error state to email unavailable',
+                target: '#Sign up.Idle',
+            },
+            'Email invalid': {
+                actions: 'Set error state to email is invalid',
+                target: '#Sign up.Idle',
+            },
+            'Weak password': {
+                actions: 'Set error state to password is weak',
+                target: '#Sign up.Idle',
+            },
+            'Unknown error': {
+                actions: ['Trigger unknown error toast'],
+                target: '#Sign up.Idle',
+            },
+        },
+    });
+
+const AuthenticationSignUpFormScreen: React.FC<SignUpFormScreenProps> = ({
+    navigation,
+}) => {
+    const {
+        control,
+        handleSubmit,
+        formState: { errors },
+        setError,
+    } = useForm<AuthenticationSignUpFormFormFieldValues>();
+
+    const sx = useSx();
+    // const { appService } = useAppContext();
+
+    const [_state, send] = useMachine(signUpMachine, {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏When I instantiate a machine without needing its current state, I like to use `useInterpret` instead of `useMachine` as `useMachine` makes the component rerender each time the state changes.